ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -949,6 +949,7 @@ public MessageInfo findKey(StoreKey key) throws StoreException {
         messageInfoMap.put(storeKey, findKey(storeKey));
       } catch (StoreException e) {
         logger.error("findKey failed for key {} with reason {}", storeKey.getID(), e.getErrorCode().toString());
+        throw e;
```

Comment:
if we are throwing exception here, then we don't need the try-catch statement.

Reply:
The reason for try-catch here is because I want to log a meaningful message with the key that resulted in exception. The exception shouldn't be thrown often here, hence I thought it might be a good idea to log it here to help with quick debugging, and it will not result in any performance issues.

Reply:
I am good with that.